bin/jruby.sh

```diff
@@ -44,70 +52,66 @@ esceval()
         [ $# -gt 0 ] || break
         output="$output $escaped"
     done
-    result="$output $escaped"
+    REPLY="$output $escaped"
 }
 
 # assign LISTNAME ELEMENT [ELEMENT...]
 #
 # Assign ELEMENT to the list named by LISTNAME.
 assign() {
     local listname="$1"
-    local result=
+    local REPLY=
     shift
 
     esceval "$@"
-    eval "$listname=\"\${result}\""
+    eval "$listname=\"\${REPLY}\""
 }
 
 # append LISTNAME ELEMENT [ELEMENT...]
 #
 # Append ELEMENT to the list named by LISTNAME.
 append() {
     local listname="$1"
-    local result=
+    local REPLY=
     shift
 
     esceval "$@"
-    eval "$listname=\"\${$listname} \${result}\""
+    eval "$listname=\"\${$listname} \${REPLY}\""
 }
 
 # prepend LISTNAME ELEMENT [ELEMENT...]
 #
 # Prepend ELEMENT to the list named by LISTNAME, preserving order.
 prepend() {
     local listname="$1"
-    local result=
+    local REPLY=
     shift
 
     esceval "$@"
-    eval "$listname=\"\${result} \${$listname}\""
-}
-
-# map FUNCNAME LISTNAME1 LISTNAME2
-#
-# Run FUNCNAME on LISTNAME1 and the list named by LISTNAME2
-map() {
-    local funcname="$1"
-    local listname="$2"
-    # Load contents of array named by $3
-    eval eval set -- "\"\${$3}\""
-    "$funcname" "$listname" "$@"
+    eval "$listname=\"\${REPLY} \${$listname}\""
 }
 
 # extend LISTNAME1 LISTNAME2
 #
 # Append the elements stored in the list named by LISTNAME2
 # to the list named by LISTNAME1.
 extend() {
-    map append "$@"
+    eval "$1=\"\${$1} \${$2}\""
 }
 
 # preextend LISTNAME1 LISTNAME2
 #
 # Prepend the elements stored in the list named by LISTNAME2
 # to the named by LISTNAME1, preserving order.
 preextend() {
-    map prepend "$@"
+    eval "$1=\"\${$2} \${$1}\""
+}
+
+# echo [STRING...]
+#
+# Dumb echo, i.e. print arguments joined by spaces with no further processing
+echo() {
```

Comment:
Functions like this will only be visible within this script, correct?

Reply:
After chatting with @mrnoname1000 and trying some subprocess checks, it appears the functions defined in this launcher script do not propagate beyond the script:

```
$ jruby -e 'system "which esceval"'
$ jruby -e 'system "command -v esceval"'
$
```

Reply:
Yeah, unless it's run in bash and exported with `export -f echo` or a preceding `set -a` it won't inherit via the environment.